test/e2e/network/service.go

```diff
@@ -2027,6 +2027,225 @@ var _ = common.SIGDescribe("Services", func() {
+		jig := e2eservice.NewTestJig(cs, ns, serviceName)
+		jig.ExternalIPs = true
+
+		local := v1.ServiceInternalTrafficPolicyLocal
+		svc, err := jig.CreateTCPService(func(svc *v1.Service) {
+			svc.Spec.Ports = []v1.ServicePort{
+				{Port: 80, Name: "http", Protocol: v1.ProtocolTCP, TargetPort: intstr.FromInt(80)},
+			}
+			svc.Spec.InternalTrafficPolicy = &local
+		})
+		framework.ExpectNoError(err)
+		defer func() {
+			framework.Logf("Cleaning up the webserver service")
+			err := cs.CoreV1().Services(ns).Delete(context.TODO(), serviceName, metav1.DeleteOptions{})
+			framework.ExpectNoError(err, "failed to delete service: %s in namespace: %s", serviceName, ns)
+		}()
+		serviceIP := svc.Spec.ClusterIP
+		framework.Logf("internal-traffic-policy-test cluster ip: %s", serviceIP)
+		ginkgo.By("Picking 2 Nodes to test whether internal traffic policy is respected")
+		nodes, err := e2enode.GetBoundedReadySchedulableNodes(cs, 2)
+		framework.ExpectNoError(err)
+		nodeCounts := len(nodes.Items)
+		if nodeCounts < 2 {
+			e2eskipper.Skipf("The test requires at least two ready nodes on %s, but found %v", framework.TestContext.Provider, nodeCounts)
+		}
+
+		ginkgo.By("Creating 2 webserver pods to be part of the TCP service")
+		webserverPod0 := e2epod.NewAgnhostPod(ns, "echo-hostname-0", nil, nil, nil, "netexec", "--http-port", strconv.Itoa(servicePort))
+		webserverPod0.Labels = jig.Labels
+		webserverPod0.Spec.NodeName = nodes.Items[0].Name
+		_, err = cs.CoreV1().Pods(ns).Create(context.TODO(), webserverPod0, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+		framework.ExpectNoError(e2epod.WaitTimeoutForPodReadyInNamespace(f.ClientSet, webserverPod0.Name, f.Namespace.Name, framework.PodStartTimeout))
+
+		webserverPod1 := e2epod.NewAgnhostPod(ns, "echo-hostname-1", nil, nil, nil, "netexec", "--http-port", strconv.Itoa(servicePort))
+		webserverPod1.Labels = jig.Labels
+		webserverPod1.Spec.NodeName = nodes.Items[1].Name
+		_, err = cs.CoreV1().Pods(ns).Create(context.TODO(), webserverPod1, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+		framework.ExpectNoError(e2epod.WaitTimeoutForPodReadyInNamespace(f.ClientSet, webserverPod1.Name, f.Namespace.Name, framework.PodStartTimeout))
+		defer func() {
+			framework.Logf("Cleaning up the echo server pod")
+			err := cs.CoreV1().Pods(ns).Delete(context.TODO(), webserverPod0.Name, metav1.DeleteOptions{})
+			framework.ExpectNoError(err, "failed to delete pod: %s on node", webserverPod0.Name)
+			err = cs.CoreV1().Pods(ns).Delete(context.TODO(), webserverPod1.Name, metav1.DeleteOptions{})
+			framework.ExpectNoError(err, "failed to delete pod: %s on node", webserverPod1.Name)
+		}()
+
+		validateEndpointsPortsOrFail(cs, ns, serviceName, portsByPodName{
+			webserverPod0.Name: {servicePort},
+			webserverPod1.Name: {servicePort},
+		})
+
+		ginkgo.By("Creating 2 pause pods that will connect to the webservers")
+		pausePod0 := &v1.Pod{
```

Comment:
why not using ` e2epod.NewAgnhostPod` ? by default it generates a pause pod

Reply:
because it doesn't let me override NodeName :)

Reply:
:thinking: you are doing it here https://github.com/kubernetes/kubernetes/pull/103409#discussion_r662588671

Reply:
Oh I remember now, it's because NewAgnhostPod hardcodes the container to `agnhost-container` but I want `agnhost-pause` her

Reply:
I think that it does that if there are no arguments, it gives you a pause container 

https://github.com/kubernetes/kubernetes/blob/1bd00776b5d78828a065b5c21e7003accc308a06/test/e2e/framework/pod/resource.go#L426-L452

```
 pausePod0 := e2epod.NewAgnhostPod(f.Namespace.Name, "agnhost-pod", nil, nil, nil)
```